components/logged_in/index.js

```diff
@@ -30,7 +30,15 @@ function mapStateToProps(state, ownProps) {
 }
 
 // NOTE: suggestions where to keep this welcomed
-const getChannelURLAction = (channel, teamId) => (dispatch, getState) => browserHistory.push(getChannelURL(getState(), channel, teamId));
+const getChannelURLAction = (channel, teamId, url) => (dispatch, getState) => {
+    const state = getState();
+
+    if (url) {
+        return browserHistory.push(url);
```

Comment:
we should check that the url belongs to the server (does it contain the SiteUrl? or make sure it is a relative path) otherwise a crafted message might try to navigate the desktop to a different location.

we should also check that it is a valid url, we don't want to add anything random to the browserhistory

Reply:
I think it would have to be relative to prevent issues on the community cluster. At the same time, I'm wondering if subpaths might cause problems here too.

Regarding the desktop app navigating to weird places, where do these URLs come from specifically? I wouldn't be too worried if they're just sent by something we control like the desktop app or the server. Even if they could come from a web app plugin, someone writing a malicious plugin could just change `window.location` directly

Reply:
The url is a permalink so we could maybe `/^\/pl\/\w+$/.test(url)`?

Reply:
Note: `history.push` follows the same origin policy, so you cannot push external URLs, it would result in an error
so that part is covered from the browser.
Nevertheless I'm adding a check to ensure url is ours and a permalink one.

Reply:
if we check directly for it being a relative permalink, then I have no concerns :)